### DISCUSSION TITLE
timestamp in identifiers

### DISCUSSION THREAD

**Discussion by jtauber** [bible-technology/scripture-burrito#235] [Archive]
https://github.com/bible-technology/scripture-burrito/issues/208#issuecomment-747569018 says the timestamp should be required on identifiers (and I've created #234 to track the actual spec change).

But what is this the timestamp of?

Is it the time that particular authority minted that particular identifier + revision combination?

If not, what?

**Comment by jtauber** [comment]
See #227

**Reply by FoolRunning** [reply]
What I recall is that the timestamp in the identifier is the timestamp when the revision was created. The [dateCreated in the Meta section](https://docs.burrito.bible/en/v0.2.0-beta/schema_docs/source_meta.html#meta-default) is the timestamp for when the SB was created.

**Reply by jtauber** [reply]
If it's just revision timestamp as opposed to the the SB timestamp, why not put it in the same part as the SB timestamp? In particular, why is it authority-specific?

What do I do if, as an authority, I just want to add my identifier but without making any other changes? What do I use for the timestamp then?

**Reply by jonathanrobie** [reply]
The authority creates revisions, and the revisions are associated with the time and date when they were created.  For instance, Paratext version 123 was created today at 9:23 a.m. EST.  

Three SBs have been created since then, they correspond to this revision.  What would the SB timestamp be for each of these SBs, the time that the SB was created or the time of the Paratext revision?

**Comment by jtauber** [comment]
What happens if I make a change to the SB, thereby invalidating other authorities' timestamps?

**Reply by jtauber** [reply]
I've suggested in #232 and #233 that if a change is made to the SB by one authority, the identifiers of all other authorities should (must?) be removed.

**Reply by jonathanrobie** [reply]
SBs are for exchanging data.  If someone edits a SB during the exchange, all bets are off, you are no longer exchanging the intended data.  If you edit a SB, it is no longer the same SB.

That is also important for referential integrity.  If two SBs have the same identifier and revision, they must be identical.

**Reply by jtauber** [reply]
Or put another way: you can't edit an SB, only the resource underlying the SB.

**Reply by jtauber** [reply]
this is now covered in #246
